src/hvf/src/lib.rs

```diff
@@ -206,11 +209,39 @@ pub fn vcpu_set_vtimer_mask(vcpuid: u64, masked: bool) -> Result<(), Error> {
     }
 }
 
-pub struct HvfNestedBindings {
-    hv_vm_config_get_el2_supported:
-        libloading::Symbol<'static, unsafe extern "C" fn(*mut bool) -> hv_return_t>,
-    hv_vm_config_set_el2_enabled:
-        libloading::Symbol<'static, unsafe extern "C" fn(hv_vm_config_t, bool) -> hv_return_t>,
+pub fn check_nested_virt() -> Result<bool, Error> {
+    // Only M3 and newer chips will support nested virtualization. Query the
+    // system to verify if the chip has support.
+    let get_el2_supported: libloading::Symbol<
+        'static,
+        unsafe extern "C" fn(*mut bool) -> hv_return_t,
+    > = unsafe {
+        HVF.get(b"hv_vm_config_get_el2_supported")
+            .map_err(Error::FindSymbol)?
+    };
+    let mut el2_supported: bool = false;
+    let ret = unsafe { (get_el2_supported)(&mut el2_supported) };
+    if ret != HV_SUCCESS {
+        error!("processor does not support the nested virtualization functionality");
+        return Err(Error::NestedCheck);
+    }
+
+    // nested virtualization requires macOS 15.0+
+    match std::process::Command::new("sw_vers")
```

Comment:
Instead of checking the macOS version, we should simply return `false` above if we can't find the `hv_vm_config_get_el2_supported` symbol.

Reply:
Fixed, thanks!